app/src/main/java/com/team3663/scouting_app/data/MatchTypes.java

```diff
@@ -0,0 +1,105 @@
+    public void addMatchTypeRow(String in_id, String in_short_form, String in_description) {
+        matchType_list.add(new MatchTypeRow(in_id, in_short_form, in_description));
+    }
+
+    // Member Function: return the size of the list
+    public int size() {
+        return matchType_list.size();
+    }
+
+    // Member Function: Get back the Id for a given match type entry (needed for logging)
+    public String getMatchTypeShortForm(int in_id) {
+        String ret = "";
+
+        // Loop through the match type list to find a matching description and return the id
+        for (MatchTypeRow mtr : matchType_list) {
+            if (mtr.id == in_id) {
+                ret = mtr.short_form;
+                break;
+            }
+        }
+        return ret;
+    }
+
+    // Member Function: Return a string list of all records
+    public ArrayList<String> getDescriptionList() {
+        ArrayList<String> descriptions = new ArrayList<>();
+
+        for (int i = 0; i < matchType_list.size(); i++) {
+            descriptions.add(matchType_list.get(i).description);
+        }
+        return descriptions;
+    }
+
+    // Member Function: Get back the Description for a given match type entry
+    public String getMatchTypeDescription(int in_id) {
+        String ret = "";
+
+        // Loop through the match type list to find a matching id and return the description
+        for (MatchTypeRow mtr : matchType_list) {
+            if (mtr.id == in_id) {
+                ret = mtr.description;
+                break;
+            }
+        }
+        return ret;
+    }
+
+    // Member Function: Get back the Id for a given match type entry
+    public int getMatchTypeId(String in_description) {
+        int ret = 0;
+
+        // Loop through the match type list to find a matching description and return the id
+        for (MatchTypeRow mtr : matchType_list) {
+            if (mtr.description.equals(in_description)) {
+                ret = mtr.id;
+                break;
+            }
+            if (mtr.short_form.equals(in_description)) {
+                ret = mtr.id;
+                break;
+            }
+        }
+        return ret;
+    }
+
+    // Member Function: Empties out the list
+    public void clear() {
+        matchType_list.clear();
+    }
+
+    // =============================================================================================
+    // Class:       MatchTypeRow
+    // Description: Defines a structure/class to hold the information for each match type
+    // =============================================================================================
+    private static class MatchTypeRow {
+        private final int id;
+        private final String short_form;
+        private final String description;
+
+        // Constructor with individual data
+        public MatchTypeRow(String in_id, String in_short_form, String in_description) {
```

Comment:
Why is in_id a `String` and not a `int`? You immediately convert it to an `int` so why not just take in an int and make it simpler?

> Line 18
> `public void addMatchTypeRow(String in_id, String in_short_form, String in_description) {`


_(and then also make this in_id an `int` too)_

Reply:
It's to make the loaddata fn() generic with no knowledge of what the file specific data it loaded.  It was like this before my check in.  
All conversions happen in the function calls.

Reply:
Ok